test/mongodb.test.js

```diff
@@ -64,6 +64,33 @@ describe('lazyConnect', function() {
       }
     });
   });
+
+  it('should reconnect on execute when disconnected (lazyConnect = true)', function(done) {
+    var ds = getDataSource({
+      host: '127.0.0.1',
+      port: config.port,
+      lazyConnect: true,
+    });
+
+    ds.define('TestLazy', {
+      value: { type: String },
+    });
+
+    ds.connector.execute('TestLazy', 'insert', { 'value': 'test value' }, function(err, success) {
+      if (err) {
+        done(err);
+      } else {
+        var id = success.insertedIds[0];
+
+        ds.connector.disconnect();
+
+        ds.connector.execute('TestLazy', 'findOne', { _id: id }, function(err, data) {
```

Comment:
Could you update the test cases with some assertions? Expected values vs actual values.